pkg/blobplugin/main.go

```diff
@@ -83,8 +84,41 @@ func handle() {
 		klog.Fatalf("failed to get Azure Cloud Provider, error: %v", err)
 	}
 	klog.V(2).Infof("cloud: %s, location: %s, rg: %s, VnetName: %s, VnetResourceGroup: %s, SubnetName: %s", cloud.Cloud, cloud.Location, cloud.ResourceGroup, cloud.VnetName, cloud.VnetResourceGroup, cloud.SubnetName)
-
-	driver := blob.NewDriver(&driverOptions, kubeClient, cloud)
+	authProvider, err := azclient.NewAuthProvider(&cloud.ARMClientConfig, &cloud.AzureAuthConfig.AzureAuthConfig)
+	if err != nil {
+		klog.Fatalf("failed to create auth provider: %v", err)
```

Comment:
originally the driver supports empty auth, what if user does not provide cloud provider config?

Reply:
the code has changed. the clientFactory is nil if cred is nil.